samples/Rewrite/ConditionParser/ConditionRegexParser.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Rewrite.ConditionParser
+{
+    public static class ConditionRegexParser
+    {
+        private const char Not = '!';
+        private const char Dash = '-';
+        private const char Less = '<';
+        private const char Greater = '>';
+        private const char EqualSign = '=';
+
+        // Given a Condition Regex Expression, obtain a ConditionRegexStatement 
+        public static GeneralExpression ParseCondition(string condition)
+        {
+            if (condition == null)
+            {
+                condition = String.Empty;
+            }
+            var context = new ConditionParserContext(condition);
+            var results = new GeneralExpression();
+            if (!context.Next())
+            {
+                return null;
+            }
+
+            if (context.Current == Not)
+            {
+                results.Invert = true;
+                if (!context.Next())
+                {
+                    return null;
+                }
+            }
+
+            if (context.Current == Greater)
+            {
+                if (!context.Next())
+                {
+                    throw new FormatException(); 
+                }
+                if (context.Current == EqualSign)
+                {
+                    if (!context.Next())
+                    {
+                        throw new FormatException();
+                    }
+                    results.Operation = OperationType.GreaterEqual;
+                    results.Type = ConditionType.StringComp;
+                } 
+                else
+                {
+                    results.Operation = OperationType.Greater;
+                    results.Type = ConditionType.StringComp;
+                }
+            }
+            else if (context.Current == Less)
+            {
+                if (!context.Next())
+                {
+                    throw new FormatException();
+                }
+                if (context.Current == EqualSign)
+                {
+                    if (!context.Next())
+                    {
+                        throw new FormatException();
+                    }
+                    results.Operation = OperationType.LessEqual;
+                    results.Type = ConditionType.StringComp;
+                }
+                else
+                {
+                    results.Operation = OperationType.Less;
+                    results.Type = ConditionType.StringComp;
+                }
+            }
+            else if (context.Current == EqualSign)
+            { 
+                if (!context.Next())
+                {
+                    throw new FormatException();
+                }
+                results.Operation = OperationType.Equal;
```

Comment:
If I were to merge them, I would need separate ones for integer comparison and string comparison. Like IntGreater, StringGreater, etc